tests/src/application/validation/validators/name-validation.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { InvalidParamError } from '@/application/errors'
+import { NameValidation, type NameValidator } from '@/application/validation'
+
+interface SutTypes {
+  sut: NameValidation
+  nameValidatorStub: NameValidator
+}
+
+const makeNameValidatorStub = (): NameValidator => {
+  class NameValidatorStub implements NameValidator {
+    isValid (name: string): boolean {
+      return true
+    }
+  }
+  return new NameValidatorStub()
+}
+
+const makeSut = (): SutTypes => {
+  const fakeFieldName: string = 'fieldName'
+  const nameValidatorStub = makeNameValidatorStub()
+  const sut = new NameValidation(fakeFieldName, nameValidatorStub)
+  return {
+    sut,
+    nameValidatorStub
+  }
+}
+
+describe('NameValidation', () => {
+  it('should returns invalidParamError if fieldName is not a string', () => {
```

Comment:
I usualy group tests by dependency. For example, `NameValidation` has two dependencies, `fieldName` and `nameValidator`, so I think that is interesting make two groups. You can create this "dependencies groups" with `describe('NameOfGroup', => {...})`. It's not a strict rule, but usualy when we use the Stub, like `nameValidatorStub`, we are in group of `nameValidator`. What you think?